index/scorch/segment/zap/intcoder.go

```diff
@@ -134,3 +139,35 @@ func (c *chunkedIntCoder) Write(w io.Writer) (int, error) {
 func (c *chunkedIntCoder) FinalSize() int {
 	return len(c.final)
 }
+
+// chunkLengthsToOffsets converts the chunk length array
+// to a chunk starting offset array. The readChunkBoundary
+// will figure out the start and end of every chunk from
+// these offsets. Starting offset of i'th index is stored
+// in i-1'th position except for 0'th index and ending offset
+// is stored at i'th index position.
+// For 0'th element, starting position is always zero.
+// eg:
+// Lens ->  5 5 5 5 => 5 10 15 20
+// Lens ->  0 5 0 5 => 0 5 5 10
+// Lens ->  0 0 0 5 => 0 0 0 5
+// Lens ->  5 0 0 0 => 5 5 5 5
+// Lens ->  0 5 0 0 => 0 5 5 5
+// Lens ->  0 0 5 0 => 0 0 5 5
+func chunkLengthsToOffsets(lengths []uint64) {
+	var runningOffset uint64
+	var index, i int
+	for i = 1; i <= len(lengths); i++ {
+		runningOffset += lengths[i-1]
```

Comment:
wondering if the same approach from PR https://github.com/blevesearch/bleve/pull/821 should be applied here -- instead of needing this additional fixup step?